CrowsZA/functions/fn_surrenderChance.sqf

```diff
@@ -0,0 +1,137 @@
+/*/////////////////////////////////////////////////
+Author: Landric
+			   
+File: fn_surrenderChance.sqf
+Parameters:
+	unit					- (unit)
+	confrontationRadius		- (float) radius within which pointing a weapon at the unit will trigger a response
+	surrenderChance			- (float) min: 0 max: 1
+	holdFire				- (bool) will unit hold fire until confronted
+	hesitation				- (float) maximum time unit will stall before acting once confronted
+
+Return: bool        		- True if method ran successfully, false if aborted (i.e. bad params were passed)
+
+// This code inspired by CQB Interactions By Flex7103
+// https://steamcommunity.com/sharedfiles/filedetails/?id=2942202773
+
+*///////////////////////////////////////////////
+params [
+	["_unit", objNull, [objNull]],
+	["_confrontationRadius", 15, [15]],
+	["_surrenderChance", 0.33, [0.33]],
+	["_holdFire", true, [true]],
+	["_hesitation", 0, [0]]
+];
+
+
+if(isNull _unit) exitWith { false };
+
+
+// TODO: Add check if unit has already had this module applied/is already surrendered/etc
+
+
+if(_holdFire) then {
+	_unit setUnitCombatMode "BLUE";
+	_unit setCombatBehaviour "CARELESS";
+};
+
+
+private _trigger = createTrigger ["EmptyDetector", getPos _unit, false];
+_trigger setTriggerArea [_confrontationRadius, _confrontationRadius, 0, false];
+_trigger setTriggerActivation ["ANYPLAYER", "PRESENT", true];
+
+_trigger setVariable ["_unit", _unit];
+_unit setVariable ["_trigger", _trigger];
+_unit setVariable ["_hesitation", _hesitation];
+_unit setVariable ["_surrenderChance", _surrenderChance];
+
+_trigger attachTo [_unit];
+
+
+// TODO: Optionally ensure that the player pointing a weapon is considered HOSTILE to the units side
+// (don't mandate - could be useful to encourage friendly units, e.g. CIVs, to surrender too)
+
+private _condition = "
+if(!this) exitWith { false };
+private _unit = thisTrigger getVariable ""_unit"";
+{
+	[_unit, {
+		missionNamespace setVariable [
+			""LND_unit_callback"",
```

Comment:
Should probably start with the "crowsza" tag to identify it is from this addon and ensure there is no conflicts. So you can do "crowsza_LND_....".